src/main/java/com/google/ehub/data/CommentDataManager.java

```diff
@@ -20,16 +21,18 @@ public void addItemComment(String message, long timestamp){
 
   }
 
-  public ArrayList<CommentData> retrieveItemComment(){
+  public ArrayList<CommentData> retrieveItemComment(long itemId){
     Query itemCommentQuery = new Query("ItemPageComments").addSort("timestamp",SortDirection.ASCENDING);
     DatastoreService itemCommentDatastoreRetrieval = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery itemCommentResults = itemCommentDatastoreRetrieval.prepare(itemCommentQuery);
     ArrayList<CommentData> results = new ArrayList<>();
     for (Entity entity : itemCommentResults.asIterable()) {
+      if(entity.getProperty("itemId") == itemId){
```

Comment:
A better way to query for comments with a specific Id would be to use a Filter predicate for the query that checks the equality of the Id and returns a single entity.
Example:
```
Query q =
    new Query("Person")
        .setFilter(new FilterPredicate("lastName", FilterOperator.EQUAL, targetLastName));

PreparedQuery pq = datastore.prepare(q);
Entity result = pq.asSingleEntity();
```